SIMDString.h

```diff
@@ -801,7 +801,7 @@ class
             return append(c);
         }
         replace(pos - m_data, 0, 1, c);
-        return m_data + pos;
+        return pos;
```

Comment:
`pos` here is an iterator (aka `const char*`), so it cannot be added to another `const char*` (`m_data`).

IMHO it is a logical error (and also a compile error).

Reply:
This has been fixed as of https://github.com/RobloxResearch/SIMDString/commit/849525764519d973e2976e947dc30868f3a9c5c0